dotnet/src/Experimental/Process.Abstractions/Models/KernelProcessStepStateMetadata.cs

```diff
@@ -8,6 +8,9 @@ namespace Microsoft.SemanticKernel.Process.Models;
 /// <summary>
 /// Step state used for State Persistence serialization
 /// </summary>
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "$type", UnknownDerivedTypeHandling = JsonUnknownDerivedTypeHandling.FallBackToNearestAncestor)]
+[JsonDerivedType(typeof(KernelProcessStepStateMetadata), typeDiscriminator: nameof(KernelProcessStepStateMetadata))]
```

Comment:
(Wondering) Considering the scope of this hierarachy (closed system / limited number of types), I might consider a more concise type discriminator that isn't bound to the type name: `Process`, `Step`, `Map` ?  (You are already aliasing property names so they aren't directly bound to model naming.)

Ultimately, either approach is just fine.

Reply:
added changes in last commit to change the type discriminator